README.md

```diff
@@ -146,6 +148,106 @@ title,content
 My Title,Hello World!
 ```
 
+### Creating a custom Task
+
+TODO: Add generation instructions
+
+If you have a special use case requiring iteration over an unsupported
+collection type, such as external resources fetched from some API, you can
+implement the `enumerator_builder` method instead.
+
+This method should return an object responding to `enumerator(context:)` with
+an Enumerator, yielding pairs of `[item, item_cursor]` to the `process` method
+your Task defines. In order for your enumerator to support resuming iteration
+part way through, you may use the `context.cursor`.
+
+You may optionally provide an implementation for `count`, if appropriate.
+
+```ruby
+# app/tasks/maintenance/shopping_list_task.rb
+module Maintenance
+  class ShoppingListTask < MaintenanceTasks::Task
+    def enumerator_builder
+      IngredientsEnumerator.new
+    end
+
+    def process(ingredient)
+      ShoppingList.add(ingredient)
+    end
+
+    class IngredientEnumerator
+      def enumerator(context:)
+        Enumerator.new do |yielder|
+          cursor = context.cursor
+
+          if cursor.nil?
+            recipe = FancyRecipeAPI.random_recipe
+            ingredient_id = nil
+          else
+            recipe_id, ingredient_id = cursor.split(':', 2)
+            recipe = FancyRecipeAPI.recipe(recipe_id)
+          end
+
+          loop do
+            page = if ingredient_id.nil?
+              FancyRecipeAPI
+                .paginated_ingredients(recipe.id, max: 5)
+            else
+              FancyRecipeAPI
+                .paginated_ingredients(recipe.id, max: 5, after: ingredient_id)
+            end
+
+            page.entries.each do |ingredient|
+              ingredient_id = ingredient.id
+              cursor = "#{recipe.id}:#{ingredient.id}"
+
+              yielder.yield([ingredient, cursor])
+            end
+
+            break unless page.has_next?
+          end
+        end
+      end
+    end
+  end
+end
+```
+
+In some cases, you may have no use for a cursor (e.g. iterating over some
+collection until it is empty), in which case your Enumerator may yield `nil`
```

Comment:
```suggestion
In some cases, you may have no use for a cursor (e.g. taking items off a queue
until it is empty), in which case your Enumerator may yield `nil`
```
Maybe `queue` instead of `collection` to make things clearer?